LEDControl/LEDControl.cpp

```diff
@@ -0,0 +1,291 @@
+                    success = true;
+                } else {
+                        LOGERR("dsFPGetSupportedLEDStates returned error %d", err);
+                }
+
+            } catch (...){
+                LOGERR("Exception in supportedLEDStates");
+            }
+            setResponseArray(response, "supportedLEDStates", supportedLEDStates);
+            returnResponse(success);
+        }
+
+        /**
+         * @brief This method returns current led state of the platform.
+         *
+         * @param: None.
+         * @return Returns the success code of underlying method.
+         */
+        uint32_t LEDControl::getLEDState(const JsonObject& parameters, JsonObject& response)
+        {
+            bool success = false;
+
+            try
+            {
+                dsFPDLedState_t state;
+               dsError_t err = dsFPGetLEDState (&state);
+               if (!err) {
+                    success = true;
+                    switch (state) {
+                    case dsFPD_LED_DEVICE_NONE:
+                        response["state"] = FPD_LED_DEVICE_NONE;
+                        break;
+                    case dsFPD_LED_DEVICE_ACTIVE:
+                        response["state"] = FPD_LED_DEVICE_ACTIVE;
+                        break;
+                    case dsFPD_LED_DEVICE_STANDBY:
+                        response["state"] = FPD_LED_DEVICE_STANDBY;
+                        break;
+                    case dsFPD_LED_DEVICE_WPS_CONNECTING:
+                        response["state"] = FPD_LED_DEVICE_WPS_CONNECTING;
+                        break;
+                    case dsFPD_LED_DEVICE_WPS_CONNECTED:
+                        response["state"] = FPD_LED_DEVICE_WPS_CONNECTED;
+                        break;
+                    case dsFPD_LED_DEVICE_WPS_ERROR:
+                        response["state"] = FPD_LED_DEVICE_WPS_ERROR;
+                        break;
+                    case dsFPD_LED_DEVICE_FACTORY_RESET:
+                        response["state"] = FPD_LED_DEVICE_FACTORY_RESET;
+                        break;
+                    case dsFPD_LED_DEVICE_USB_UPGRADE:
+                        response["state"] = FPD_LED_DEVICE_USB_UPGRADE;
+                        break;
+                    case dsFPD_LED_DEVICE_SOFTWARE_DOWNLOAD_ERROR:
+                        response["state"] = FPD_LED_DEVICE_SOFTWARE_DOWNLOAD_ERROR;
+                        break;
+
+                    default :
+                        LOGERR("Unsupported LEDState %d", state);
+                        success = false;
+                        break;
+                    }
+                } else {
+                    LOGERR("dsFPGetLEDState returned error %d", err);
+                }
+            }
+            catch(...)
+            {
+                LOGERR("Exception in dsFPGetLEDState");
+                success = false;
```

Comment:
Can we handle returning error object here. Pls take the time and handle all error scenarios for all the new APIs.